src/viam/sdk/common/service_helper.cpp

```diff
@@ -0,0 +1,42 @@
+#include <viam/sdk/common/service_helper.hpp>
+
+#include <sstream>
+
+namespace viam {
+namespace sdk {
+
+::grpc::Status ServiceHelperBase::fail(::grpc::StatusCode code, const char* message) const noexcept
+    try {
+    std::ostringstream stream;
+    stream << '[' << method_ << "]: " << message;
+    return {code, stream.str()};
+} catch (...) {
+    return {code, message};
+}
+
+::grpc::Status ServiceHelperBase::failNoRequest() const noexcept {
+    return fail(::grpc::INVALID_ARGUMENT, "Called without a `request` object");
+}
+
+::grpc::Status ServiceHelperBase::failNoResource(const std::string& name) const noexcept try {
+    std::ostringstream stream;
+    stream << "Failed to find resource `" << name << "`";
+    return fail(::grpc::INVALID_ARGUMENT, stream.str().c_str());
```

Comment:
I wonder a bit if this should actually be `::grpc::NOT_FOUND`.

Reply:
More believable than `INVALID_ARGUMENT` imo. We certainly have not standardized what situations should result in what error codes across SDKs.

Reply:
agreed, `NOT_FOUND` seems more intuitive.

Reply:
Done